src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -692,6 +708,16 @@ protected virtual void Dispose(bool disposing)
                 this.disposed = true;
                 if (disposing)
                 {
+                    if (this.eventReceivers != null)
+                    {
+                        foreach (var receiver in this.eventReceivers)
+                        {
+                            receiver.Dispose();
+                        }
+
+                        this.eventReceivers.Clear();
```

Comment:
A little surprised you didn't set this to null instead of just clearing it

Reply:
Good point. Fixed.